braindecode/datasets/base.py

```diff
@@ -368,60 +374,73 @@ def target_transform(self, fn):
             raise TypeError('target_transform must be a callable.')
         self._target_transform = fn
 
-    def save(self, path, overwrite=False):
-        """Save dataset to files.
+    def _outdated_save(self, path, overwrite=False):
+        """This is a copy of the old saving function, that had inconsistent
+        functionality for BaseDataset and WindowsDataset. It only exists to
+        assure backwards compatibility by still being able to run the old tests.
+
+        Save dataset to files.
 
         Parameters
         ----------
         path : str
             Directory to which .fif / -epo.fif and .json files are stored.
         overwrite : bool
-            Whether to delete old files (.json, .fif, -epo.fif) in specified directory
-            prior to saving.
+            Whether to delete old files (.json, .fif, -epo.fif) in specified
+            directory prior to saving.
         """
-        assert len(self.datasets) > 0, "Expect at least one dataset"
-        assert (hasattr(self.datasets[0], 'raw') + hasattr(
-            self.datasets[0], 'windows') == 1), (
-            "dataset should have either raw or windows attribute")
+        warnings.warn('This function only exists for backwards compatibility '
+                      'purposes. DO NOT USE!', UserWarning)
+        if len(self.datasets) == 0:
+            raise ValueError("Expect at least one dataset")
+        if not (hasattr(self.datasets[0], 'raw') or hasattr(
+                self.datasets[0], 'windows')):
+            raise ValueError("dataset should have either raw or windows "
+                             "attribute")
         file_names_ = ["{}-raw.fif", "{}-epo.fif"]
```

Comment:
I think this part becomes a bit confusing with `file_names_` `file_names` etc. suggest to rename:
`file_names_` to `file_templates` or `file_name_templates`. And also further below change `file_name` to `file_template` or `file_name_template` @gemeinl